src/main/java/com/rabbitmq/client/ConnectionFactory.java

```diff
@@ -377,6 +382,65 @@ private static String uriDecode(String s) {
         }
     }
 
+    /**
+     * Convenience method for setting some fields from query parameters
+     * Will handle only a subset of the query parameters supported by the
+     * official erlang client
+     * https://www.rabbitmq.com/uri-query-parameters.html
+     * @param rawQuery is the string containing the raw query parameters part from a URI
+     */
+    private void setQuery(String rawQuery) {
+        Map<String, String> parameters = new HashMap<>();
+        
+        // parsing the query parameters
+        try {
+            for (String param : rawQuery.split("&")) {
+                String[] pair = param.split("=");
+                String key = URLDecoder.decode(pair[0], "US-ASCII");
```

Comment:
Any reason to not use `UTF-8` here?

Reply:
I used the same charset as in uriDecode line 378, which is used to decode username password and vhost.
```
    private static String uriDecode(String s) {
        try {
            // URLDecode decodes '+' to a space, as for
            // form encoding.  So protect plus signs.
            return URLDecoder.decode(s.replace("+", "%2B"), "US-ASCII");
        }
        catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
```
https://github.com/rabbitmq/rabbitmq-java-client/pull/672/files#diff-045e321e0fb12f11085653a7ed09f16518f9f21a7c80990c30720958dc17e954R378